File: app/src/main/java/com/tetsoft/typego/AchievementsFragment.java
package com.tetsoft.typego;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.tetsoft.typego.adapters.AchievementsAdapter;
import com.tetsoft.typego.utils.Achievement;
import com.tetsoft.typego.utils.User;
import java.util.ArrayList;

public class AchievementsFragment extends Fragment {


    public AchievementsFragment() {

    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_achievements, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        User currentUser = User.getFromStoredData(getContext());
        ArrayList<Achievement> staticAchievements = Achievement.getAchievementList(getContext());
        RecyclerView rvAchievements = view.findViewById(R.id.rv_achievements);
        rvAchievements.setAdapter(new AchievementsAdapter(
                getContext(),
                staticAchievements,
                currentUser));
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        rvAchievements.setLayoutManager(linearLayoutManager);
    }
}

File: app/src/main/java/com/tetsoft/typego/ui/fragment/AchievementsFragment.kt
package com.tetsoft.typego.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.hilt.navigation.fragment.hiltNavGraphViewModels
import androidx.recyclerview.widget.LinearLayoutManager
import com.tetsoft.typego.R
import com.tetsoft.typego.databinding.FragmentAchievementsBinding
import com.tetsoft.typego.ui.fragment.achievements.AchievementsViewModel

class AchievementsFragment : BaseFragment<FragmentAchievementsBinding>() {

    private val viewModel : AchievementsViewModel by hiltNavGraphViewModels(R.id.main_navigation)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding.rvAchievements.adapter = com.tetsoft.typego.adapter.achievements.AchievementsAdapter(
            requireContext(),
            viewModel.getAchievementsList(),
            viewModel.getResults(),
            viewModel.getAchievementsProgressList()
        )
        val linearLayoutManager = LinearLayoutManager(context)
        binding.rvAchievements.layoutManager = linearLayoutManager
    }

    override fun initBinding(
        inflater: LayoutInflater,
        container: ViewGroup?
    ): FragmentAchievementsBinding {
        return FragmentAchievementsBinding.inflate(inflater, container, false)
    }
}